### ISSUE TITLE
Missing Recommended Dep?

### ISSUE THREAD

**Issue by frioux** [andrewd18/df-lnp-installer#23]
I installed all the deps as recommended at the beginning of the guide and got all the way to this error:

```
     Project MESSAGE: Release Mode
     Project MESSAGE: Setting up for Linux
     Project MESSAGE: Setting up for Linux Install
     Project MESSAGE: Warning: unknown QT: widgets
     make: Entering directory `/home/frew/code/df-lnp-installer/downloads/dwarftherapist'
     /usr/lib/x86_64-linux-gnu/qt4/bin/uic ui/scriptdialog.ui -o bin/release/ui_scriptdialog.h
     /usr/lib/x86_64-linux-gnu/qt4/bin/uic ui/scannerdialog.ui -o bin/release/ui_scannerdialog.h
     /usr/lib/x86_64-linux-gnu/qt4/bin/uic ui/pendingchanges.ui -o bin/release/ui_pendingchanges.h
     /usr/lib/x86_64-linux-gnu/qt4/bin/uic ui/optionsmenu.ui -o bin/release/ui_optionsmenu.h
     /usr/lib/x86_64-linux-gnu/qt4/bin/uic ui/mainwindow.ui -o bin/release/ui_mainwindow.h
     /usr/lib/x86_64-linux-gnu/qt4/bin/uic ui/importexportdialog.ui -o bin/release/ui_importexportdialog.h
     /usr/lib/x86_64-linux-gnu/qt4/bin/uic ui/gridviewdock.ui -o bin/release/ui_gridviewdock.h
     /usr/lib/x86_64-linux-gnu/qt4/bin/uic ui/gridviewdialog.ui -o bin/release/ui_gridviewdialog.h
     /usr/lib/x86_64-linux-gnu/qt4/bin/uic ui/dwarfdetailswidget.ui -o bin/release/ui_dwarfdetailswidget.h
     /usr/lib/x86_64-linux-gnu/qt4/bin/uic ui/dwarfdetailsdock.ui -o bin/release/ui_dwarfdetailsdock.h
     /usr/lib/x86_64-linux-gnu/qt4/bin/uic ui/customprofession.ui -o bin/release/ui_customprofession.h
     /usr/lib/x86_64-linux-gnu/qt4/bin/uic ui/columneditdialog.ui -o bin/release/ui_columneditdialog.h
     /usr/lib/x86_64-linux-gnu/qt4/bin/uic ui/about.ui -o bin/release/ui_about.h
     /usr/lib/x86_64-linux-gnu/qt4/bin/uic ui/selectparentlayoutdialog.ui -o bin/release/ui_selectparentlayoutdialog.h
     /usr/lib/x86_64-linux-gnu/qt4/bin/uic ui/roledialog.ui -o bin/release/ui_roledialog.h
     /usr/lib/x86_64-linux-gnu/qt4/bin/uic ui/optimizereditor.ui -o bin/release/ui_optimizereditor.h
     g++ -c -m64 -pipe -O2 -Wall -W -D_REENTRANT -DQT_NO_DEBUG -DQT_SCRIPT_LIB -DQT_GUI_LIB -DQT_NETWORK_LIB -DQT_CORE_LIB -DQT_SHARED -I/usr/share/qt4/mkspecs/linux-g++-64 -I. -I/usr/include/qt4/QtCore -I/usr/include/qt4/QtNetwork -I/usr/include/qt4/QtGui -I/usr/include/qt4/QtScript -I/usr/include/qt4 -Iinc -Iinc/models -Iinc/grid_view -Iinc/docks -Iui -Ithirdparty/qtcolorpicker-2.6 -Ibin/release -Ibin/release -o bin/release/dfinstancelinux.o src/dfinstancelinux.cpp
     src/dfinstancelinux.cpp:23:21: fatal error: QtWidgets: No such file or directory
     compilation terminated.
     make: *** [bin/release/dfinstancelinux.o] Error 1
     make: Leaving directory `/home/frew/code/df-lnp-installer/downloads/dwarftherapist'

     df-lnp-installer.sh: Compiling Dwarf Therapist failed. See Make output above for details. Exiting.
```

I'll keep looking to see if I can figure out what the missing dep is, but if someone else knows feel free to comment/close :)

**Comment by andrewd18** [comment]
Looks like the upstream maintainer made a pretty radical change this morning:

> changeset:   578:4294379f597b
> tag:         tip
> user:        joshb
> date:        Thu Oct 03 17:39:35 2013 +0200
> summary:     v.20.6 *\* updated the project to qt 5.1.1 **

If you can install Qt5 development libraries, that's probably what you're missing. The install script assumes Qt4 (because up until last night, that's what you needed) so Qt dependency detection is going to be broken until I can either install Qt5 reliably or force a DwarfTherapist version that uses Qt4.

**Comment by andrewd18** [comment]
If you wanted to try and tackle this yourself, you will also need to edit the df-lnp-installer.sh script.

You'll need to change this function:

```
find_qmake_qt4 () {
  for name in "qmake" "qmake-qt4"; do
    if [ -n "$(which $name)" ] && [ $($name -query QT_VERSION | cut -d . -f 1) -eq 4 ]; then
        echo $name
        break
    fi
  done
}
```

to look like this:

```
find_qmake_qt4 () {
  for name in "qmake" "qmake-qt5"; do
    if [ -n "$(which $name)" ] && [ $($name -query QT_VERSION | cut -d . -f 1) -eq 5 ]; then
        echo $name
        break
    fi
  done
}
```

**Comment by frioux** [comment]
I may look into it, but for now decided to just skip on that tool entirely.

**Comment by wesQ3** [comment]
Would it be ok to explicitly declare which revision of the Dwarf Therapist repo you want to build to prevent upstream changes from unexpectedly breaking installs in the future?

I just built fine on Ubuntu 12.04 LTS with this:

``` diff
diff --git a/df-lnp-installer.sh b/df-lnp-installer.sh
index 172d999..c4d036d 100755
--- a/df-lnp-installer.sh
+++ b/df-lnp-installer.sh
@@ -402,6 +402,7 @@ download_dwarf_therapist () {
   # Check for and fix the issue I had in 0.2.0 where I used the wrong upstream URL.
   # Get the current upstream url. If the directory doesn't exist the var will contain "".
   local CURRENT_UPSTREAM_URL="$(hg paths --cwd $DWARF_THERAPIST_HG_DIR | grep default | cut -d" " -f3)"
+  local DT_REVISION="4ef8173a7a94"

   if [ "$CURRENT_UPSTREAM_URL" != "$SPLINTERMIND_REPO_URL" ]; then
    # Inform the user (assuming they're paying attention)
@@ -413,7 +414,7 @@ download_dwarf_therapist () {
    fi

    # Reclone.
-   hg clone "$SPLINTERMIND_REPO_URL" "$DWARF_THERAPIST_HG_DIR"
+   hg clone -r $DT_REVISION "$SPLINTERMIND_REPO_URL" "$DWARF_THERAPIST_HG_DIR"
   else
    # URL is good; just get the latest changes.
    hg update --cwd "$DWARF_THERAPIST_HG_DIR"
```

**Comment by andrewd18** [comment]
wesQ3, that's a great idea. Implemented in 0.3.1. Thank you. Since this should resolve the issue (I was able to recreate and resolve it on my box too), I'm going to close this and open a separate enhancement issue for Splintermind 20.6 and Qt5 support.
